scripts/automation/create_release_tag.py

```diff
@@ -0,0 +1,446 @@
+                    return "No history notes found"
+
+                history_content = wheel.read(history_files[0]).decode('utf-8')
+
+                # Match any line starting with the version number
+                version_pattern = rf"^{re.escape(version)}.*?\n(?:[-=+~]+\n)?(.*?)(?=^[\d.]+[a-z0-9].*?(?:\n[-=+~]+)?|\Z)"
+                match = re.search(version_pattern, history_content, re.DOTALL | re.MULTILINE)
+
+                if match:
+                    return match.group(1).strip()
+
+                return "No history notes found for this version"
+
+    except Exception as e:
+        print(f"Error getting history notes: {e}")
+        return "No history notes found for this version"
+
+
+def get_history_note_from_source(version: str, extension_name: str) -> str:
+    """Get history notes from source code HISTORY.rst"""
+    try:
+        history_path = f"src/{extension_name}/HISTORY.rst"
+        if not os.path.exists(history_path):
+            return "No history notes found"
+
+        with open(history_path, 'r', encoding='utf-8') as f:
+            history_content = f.read()
+
+        # Match any line starting with the version number
+        version_pattern = rf"^{re.escape(version)}.*?\n(?:[-=+~]+\n)?(.*?)(?=^[\d.]+[a-z0-9].*?(?:\n[-=+~]+)?|\Z)"
+        match = re.search(version_pattern, history_content, re.DOTALL | re.MULTILINE)
+
+        if match:
+            return match.group(1).strip()
+
+        return "No history notes found in source code"
+
+    except Exception as e:
+        print(f"Error reading history from source: {e}")
+        return "No history notes found for this version"
+
+
+def main():
+    try:
+        # Get added lines from git diff
+        added_lines = get_file_diff()
+        if not added_lines:
+            print("No changes found in index.json")
+            return
+
+        # Parse filenames from added lines
+        filenames = parse_filenames(added_lines)
+        commit_sha = subprocess.check_output(
+            ["git", "rev-parse", "HEAD"],
+            text=True
+        ).strip()
+        if not filenames:
+            print("No filenames found in changes")
+            shas = parse_sha256_digest(added_lines)
+            if not shas:
+                print("No sha256Digest found in changes")
+                return
+            else:
+                for sha in shas:
+                    filename, wheel_url = get_file_info_by_sha(sha)
+                    tag_name, release_title, version = generate_tag_and_title(filename)
+                    release_id, release_body, asset_id = get_release_info(tag_name)
+                    update_release_body(release_id, commit_sha, release_body, sha, tag_name) if release_id else None
+                    update_release_asset(wheel_url, asset_id, release_id) if release_id else None
+
+        print(f"Found {len(filenames)} files to process")
+        # Process each filename
+        for filename in filenames:
```

Comment:
Since multiple extensions can be released at once, multi-version GitHub release is necessary.